File: python/a013.py

import sys

def int_to_Roman(num):
    if not num:
        return "ZERO"
    val = [
        1000, 900, 500, 400,
        100, 90, 50, 40,
        10, 9, 5, 4,
        1
    ]
    syb = [
        "M", "CM", "D", "CD",
        "C", "XC", "L", "XL",
        "X", "IX", "V", "IV",
        "I"
    ]
    roman_num = ''
    i = 0
    while num > 0:
        for _ in range(num // val[i]):
            roman_num += syb[i]
            num -= val[i]
        i += 1
    return roman_num

def str_to_roman(s):
    a = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
    ans = 0 
    for i in range(len(s)-1):    
        if a[s[i]] < a[s[i+1]]:
            ans-=a[s[i]]
        else:
            ans+=a[s[i]]
    ans += a[s[-1]]
    return ans


for a in sys.stdin:
    _input = a.strip().split(" ")
    if len(_input)-1:
        print(str_to_roman(_input[0]))
        exit(0)
        print(int_to_Roman(abs(str_to_roman(_input[0])-str_to_roman(_input[1]))))

File: java/a013.java
import java.lang.Math;
import java.util.Scanner;
import java.util.HashMap;

public class a013 {
    public static int to_str(String str) {
        HashMap<String, Integer> a = new HashMap<>();
        a.put("I", 1);
        a.put("V", 5);
        a.put("X", 10);
        a.put("L", 50);
        a.put("C", 100);
        a.put("D", 500);
        a.put("M", 1000);
        int ans = 0;
        for (int i = 0; i < str.length() - 1; i++) {
            if (a.get(str.split("")[i]) < a.get(str.split("")[i + 1])) {
                ans -= a.get(str.split("")[i]);
            } else {
                ans += a.get(str.split("")[i]);
            }
        }
        ans += a.get(str.split("")[str.length() - 1]);
        return ans;
    }

    static String to_roman(int num) {
        if (num == 0){
            return "ZERO";
        }
        int[] val = { 1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1 };
        String[] syb = { "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I" };
        String roman_num = "";
        int i = 0;
        while (num > 0) {
            for (int x = 0; 0 < Math.floor(num / val[i]); x++) {
                roman_num+=syb[i];
                num -= val[i];
                x++;
            }
            i+=1;
        }
        return roman_num;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] inputs;
        while (sc.hasNext()) {
            inputs = sc.nextLine().split(" ");
            if (inputs.length == 2) {
                System.out.println(to_roman(Math.abs(to_str(inputs[0]) - to_str(inputs[1]))));
            }
        }
        sc.close();
    }
}
